cg/meta/demultiplex/hk_functions.py

```diff
@@ -0,0 +1,217 @@
+    )
+    add_sample_fastq_files_to_housekeeper(flow_cell=flow_cell, hk_api=hk_api, store=store)
+    add_demux_logs_to_housekeeper(flow_cell=flow_cell, hk_api=hk_api, demux_api=demux_api)
+
+
+def add_demux_logs_to_housekeeper(
+    flow_cell: FlowCellDirectoryData, hk_api: HousekeeperAPI, demux_api: DemultiplexingAPI
+) -> None:
+    """Add demux logs to Housekeeper."""
+    log_file_name_pattern: str = r"*_demultiplex.std*"
+    demux_log_file_paths: List[Path] = get_files_matching_pattern(
+        directory=Path(demux_api.run_dir, flow_cell.full_name), pattern=log_file_name_pattern
+    )
+
+    tag_names: List[str] = [SequencingFileTag.DEMUX_LOG, flow_cell.id]
+    for log_file_path in demux_log_file_paths:
+        try:
+            add_file_to_bundle_if_non_existent(
+                file_path=log_file_path,
+                bundle_name=flow_cell.id,
+                tag_names=tag_names,
+                hk_api=hk_api,
+            )
+            LOG.info(f"Added demux log file {log_file_path} to Housekeeper.")
+        except FileNotFoundError as e:
+            LOG.error(f"Cannot find demux log file {log_file_path}. Error: {e}.")
+
+
+def add_sample_fastq_files_to_housekeeper(
+    flow_cell: FlowCellDirectoryData, hk_api: HousekeeperAPI, store: Store
+) -> None:
+    """Add sample fastq files from flow cell to Housekeeper."""
+
+    sample_internal_ids: List[str] = get_sample_internal_ids_from_sample_sheet(
+        sample_sheet_path=flow_cell.sample_sheet_path,
+        flow_cell_sample_type=flow_cell.sample_type,
+    )
+
+    for sample_internal_id in sample_internal_ids:
+        add_bundle_and_version_if_non_existent(bundle_name=sample_internal_id, hk_api=hk_api)
+
+        sample_fastq_paths: Optional[List[Path]] = get_sample_fastqs_from_flow_cell(
+            flow_cell_directory=flow_cell.path, sample_internal_id=sample_internal_id
+        )
+
+        if not sample_fastq_paths:
+            LOG.warning(
+                f"Cannot find fastq files for sample {sample_internal_id} in {flow_cell.path}. Skipping."
+            )
+            continue
+
+        for sample_fastq_path in sample_fastq_paths:
+            store_fastq_path_in_housekeeper(
+                sample_internal_id=sample_internal_id,
+                sample_fastq_path=sample_fastq_path,
+                flow_cell=flow_cell,
+                hk_api=hk_api,
+                store=store,
+            )
+
+
+def store_fastq_path_in_housekeeper(
+    sample_internal_id: str,
+    sample_fastq_path: Path,
+    flow_cell: FlowCellDirectoryData,
+    hk_api: HousekeeperAPI,
+    store: Store,
+) -> None:
+    sample_fastq_should_be_stored: bool = check_if_fastq_path_should_be_stored_in_housekeeper(
+        sample_id=sample_internal_id,
+        sample_fastq_path=sample_fastq_path,
+        sequencer_type=flow_cell.sequencer_type,
+        flow_cell_name=flow_cell.id,
+        store=store,
+    )
+
+    if sample_fastq_should_be_stored:
```

Comment:
There has been a bug fix applied to this function which is missing here, see https://github.com/Clinical-Genomics/cg/commit/1ae9c379e29fc8054382bd258952370bb195e5f5#diff-afa9660521ae4a7f3be5393c339ace5b0741915c3b2a2c7f2bd04cf788d3ea26R311

Reply:
added the bug fix here.